From 613289200772ce1076b18fd7317fd98efeb7c7f0 Mon Sep 17 00:00:00 2001
From: Adrien Plazas <kekun.plazas@laposte.net>
Date: Tue, 23 Jul 2019 17:11:10 +0200
Subject: [PATCH 06/38] language: Keep the search entry at the bottom

This avoids the search entry to jump around as the user searches.
---
 gnome-initial-setup/pages/language/cc-language-chooser.c | 5 -----
 gnome-initial-setup/pages/language/gis-language-page.ui  | 3 ---
 gnome-initial-setup/pages/language/language-chooser.ui   | 3 ++-
 3 files changed, 2 insertions(+), 9 deletions(-)

--- a/gnome-initial-setup/pages/language/cc-language-chooser.c
+++ b/gnome-initial-setup/pages/language/cc-language-chooser.c
@@ -395,11 +395,6 @@
         gtk_widget_show (priv->filter_entry);
         gtk_widget_grab_focus (priv->filter_entry);
 
-	gtk_scrolled_window_set_policy (GTK_SCROLLED_WINDOW (priv->scrolled_window),
-					GTK_POLICY_NEVER,
-					GTK_POLICY_AUTOMATIC);
-	gtk_widget_set_valign (GTK_WIDGET (chooser), GTK_ALIGN_FILL);
-
         priv->showing_extra = TRUE;
         gtk_list_box_invalidate_filter (GTK_LIST_BOX (priv->language_list));
         g_object_notify_by_pspec (G_OBJECT (chooser), obj_props[PROP_SHOWING_EXTRA]);
--- a/gnome-initial-setup/pages/language/gis-language-page.ui
+++ b/gnome-initial-setup/pages/language/gis-language-page.ui
@@ -12,7 +12,6 @@
             <property name="visible">True</property>
             <property name="orientation">vertical</property>
             <property name="halign">center</property>
-            <property name="valign">center</property>
             <property name="spacing">18</property>
             <property name="expand" bind-source="language_chooser" bind-property="visible" bind-flags="invert-boolean|sync-create"/>
             <child>
@@ -37,7 +36,6 @@
             <property name="margin_bottom">18</property>
             <property name="width_request">400</property>
             <property name="visible">True</property>
-            <property name="valign">start</property>
           </object>
         </child>
       </object>
--- a/gnome-initial-setup/pages/language/language-chooser.ui
+++ b/gnome-initial-setup/pages/language/language-chooser.ui
@@ -8,9 +8,10 @@
     <child>
       <object class="GtkScrolledWindow" id="scrolled_window">
         <property name="visible">True</property>
+        <property name="propagate-natural-height">True</property>
         <property name="hscrollbar-policy">never</property>
-        <property name="vscrollbar-policy">never</property>
         <property name="shadow-type">in</property>
+        <property name="valign">start</property>
         <child>
           <object class="GtkListBox" id="language_list">
             <property name="visible">True</property>
